orbs/circleci/circle-wait-job

```diff
@@ -99,83 +99,96 @@ uri = URI::HTTPS.build(host: "circleci.com",
     # concurrent_build_management job).
     # Iterate on builds and regroup builds by job name in a dict:
     # Ex: { concurrent_build_management: [build1, build2], test_js: [build3], ...}
-    builds_by_job_name = Hash.new
-    builds.each_entry do |build|
-      build_job_name = build.dig("workflows", "job_name").to_s
-      builds_by_job_name[build_job_name] = builds_by_job_name[build_job_name] || []
-      builds_by_job_name[build_job_name].push(build)
+    concurrent_builds_for_current_job = builds.select do |build|
+      build.dig("workflows", "job_name").to_s == current_job
     end
 
-    # Look for builds that have a job name == current_job since the present script is expected to
-    # be called for a job whose concurrent run is not allowed
-    concurrent_builds_for_current_job = builds_by_job_name[current_job]
-
-    if !concurrent_builds_for_current_job || concurrent_builds_for_current_job.length() == 0
+    unless concurrent_builds_for_current_job.any?
       puts "No builds running job #{current_job} were found."
       return false
     end
 
-    puts "Checking which concurrent build is the oldest (from commit perspective)..."
+    puts "\nChecking which concurrent build is the oldest (based on committer_date)..."
 
     # Find build for the oldest code
-    oldestBuild = concurrent_builds_for_current_job.first
-    oldestCommitDate = DateTime.parse(oldestBuild.dig("committer_date"))
-    concurrent_builds_for_current_job.each_entry do |build|
-      puts "Comparing build #{build.dig("build_url")} with commit date #{build.dig("committer_date")}"
-      buildCommitDate = DateTime.parse(build.dig("committer_date"))
-      if buildCommitDate <= oldestCommitDate
-        puts "\tBuild is older"
-        oldestCommitDate = buildCommitDate
-        oldestBuild = build
-      else
-        puts "\tBuild is newer"
-      end
+    oldestBuild = concurrent_builds_for_current_job.min do |build1, build2|
+      DateTime.parse(build1["committer_date"]) <=> DateTime.parse(build2["committer_date"])
     end
 
-    if build_num == oldestBuild.dig("build_num")
-      puts "\nCurrent build has the oldest code, it can continue running"
-      return false
-    else
+    puts "Oldest build is #{oldestBuild["build_num"]}"
+
+    if current_build_num != oldestBuild["build_num"]
       puts "\nCurrent build has code more recent that another one running, it must wait"
       return true
     end
+
+    puts "\nCurrent build has the oldest code, it can continue running"
+
+    # There is generally around 4 seconds between a job being successful and the following job being running.
+    # It means it is possible that an older job similar to the current job just successful and that
+    # the associate targeted job has not started yet.
+    # In this case, two target jobs would run concurrently.
+    # To avoid this edge case, we ensure that no jobs similar to the current job successful a few seconds ago.
```

Comment:
**wording**
Same wording question around `successful`:
```suggestion
    # To avoid this edge case, we ensure that no jobs similar to the current job succeeded a few seconds ago.
```